app/models/trn_request.rb

```diff
@@ -68,4 +68,8 @@ def ni_number_answered?
 
     !has_ni_number.nil? && !has_ni_number_was.nil?
   end
+
+  def previous_trn_success_for_email?
+    TrnRequest.where(email:).where.not(id:, trn: nil).present?
```

Comment:
```suggestion
    TrnRequest.where(email:).where.not(id:).where.not(trn:).present?
```
I believe this is the query that we want, because we want to detect if the same email has successfully requested **a different** TRN.